### ISSUE TITLE
127.0.0.1:8080 shows no webpage but a directory tree

### ISSUE THREAD

**Issue by cuekoo** [gitbucket/gitbucket#1121]
## Issue
**Impacted version**: 3.11 (earlier versions not tested)

**Deployment mode**: `java -jar gitbucket.war`

**Problem description**:
On my Mac everything is fine. But when I switch to Ubuntu, `127.0.0.1:8080` shows no webpage but a directory tree. `java -version` gives

```
java version "1.8.0_65"
Java(TM) SE Runtime Environment (build 1.8.0_65-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.65-b01, mixed mode)
```

**Comment by McFoggy** [comment]
What is shown in this directory tree? 
Are you sure it is related to gitbucket and that you are not seing the result of an already running http server? 
Did you tested to go to this URL without launching gitbucket first?

**Comment by cuekoo** [comment]
It's the `gitbucket` directory tree. I am sure it is related to gitbucket.
```
Directory: /

data.h2.db 	362496 bytes 	Feb 24, 2016 2:44:48 PM
data.trace.db 	295153 bytes 	Nov 18, 2015 11:08:58 AM
data/ 	4096 bytes 	Apr 28, 2015 11:22:20 AM
repositories/ 	4096 bytes 	Apr 28, 2015 11:22:20 AM
version 	4 bytes 	Jan 13, 2016 11:00:02 AM
```

**Comment by McFoggy** [comment]
Works on my machine :-(

![image](https://cloud.githubusercontent.com/assets/1119660/13279596/5850c92e-dad9-11e5-9e0e-cdebe99a572c.png)

Do you see any errors when lauching gitbucket?
Can you try collecting logs: `java -jar gitbucket.war | tee console.log`
Attach the log once your server started and that you went to http://localhost:8080.
Do you have something special in your settings?

**Comment by cuekoo** [comment]
Unfortunately I didn't find anything special in the following

![gb](https://cloud.githubusercontent.com/assets/3119748/13283333/3c19cf68-db28-11e5-9836-426fca1f893f.png)

**Comment by McFoggy** [comment]
Can you please do the following trial:
```
mkdir /tmp/gitbucket-trial
cd /tmp/gitbucket-trial
wget https://github.com/gitbucket/gitbucket/releases/download/3.11/gitbucket.war
mkdir home
java -jar gitbucket.war --gitbucket.home=home | tee launch.log
```
Then navigate to http://localhost:8080 and attach to this thread the launch.log file

**Comment by lidice** [comment]
FYI
http://stackoverflow.com/a/33266113

**Comment by McFoggy** [comment]
I have the same message on my ubuntu virtual box but gitbucket works.
AFAIK gitbucket does not use JSP but:

- scalatra for page rendering
- jgit behind a webfilter for git related things
- standard servlets & filters for internal usage

**Comment by ritschwumm** [comment]
i've seen someting very similar a few days ago with one of my own apps.
it ran fine for a couple of days when suddenly i got a directory listing and
none of my content any more. without - iirc - any interesting log output.

i'm using a runner similar to gitbucket's, albeit bundled in a capsule
together with the jetty jars and a separate war file. the jetty version i
used was 9.3.7.v20160115 though.

tl;dr this looks like a jetty issue to me

**Comment by cuekoo** [comment]
@McFoggy It works in `/tmp/gitbucket-trial`! Then I found that copying my previous `gitbucket.war` to other places or simply renaming it also works. In my humble opinion, the problem was caused by a directory named `gitbucket` under the directory that `gitbucket.war` resides, and the previously shown directory tree on `localhost:8080` is exactly the `gitbucket` directory. 

[launch.log.txt](https://github.com/gitbucket/gitbucket/files/144668/launch.log.txt)

**Comment by McFoggy** [comment]
Fine that you found your problem
